### DISCUSSION TITLE
Is possible to port devilutionX to PS2?

### DISCUSSION THREAD

**Discussion by SkullEmperor** [diasurgical/DevilutionX#1892] [Ideas]
Looking in the source files I noted in /Packaging/ctr/template.rsf (configurations for 3ds port) that exist an option for run in 32Mb:
SystemMode : 80MB # 64MB(Default)/96MB/80MB/72MB/32MB)
With that in mind is possible a PS2 run a port? 
The PS2 has a 294 MHz processor (which increased to 299 MHz on later models), 32 MB RAM, a GPU clocked at 147 MHz, 4MB video RAM.

**Comment by AJenbo** [comment]
It should be possible, it was ported to Xbox which also only has 32MB of ram. For low memory devices we have an option to stream all audio instead of loading it.

If you intend to do this you may have to make some optimizations depending on how the device manages memory.

**Comment by OMightyBuggy** [comment]
Is it possible to compile the source to run on PS2? I have the PS2 SDK do I need anything on the Diablo side other than the source?

**Reply by AJenbo** [reply]
I have it mostly working here https://github.com/diasurgical/devilutionX/pull/5238

There are issues with reading the save game and audio

**Reply by OMightyBuggy** [reply]
Thank you much. :)
